src/Types/Input/EnumInputType.hack

```diff
@@ -0,0 +1,47 @@
+namespace Slack\GraphQL\Types;
+
+use namespace HH\Lib\C;
+use namespace Graphpinator\Parser\Value;
+use namespace Slack\GraphQL;
+
+
+abstract class EnumInputType extends NamedInputType {
+
+    <<__Enforceable>>
+    abstract const type TCoerced as arraykey;
+    abstract const \HH\enumname<this::TCoerced> HACK_ENUM;
+
+    <<__Override>>
+    public function coerceValue(mixed $value): this::TCoerced {
+        $enum = static::HACK_ENUM;
+        GraphQL\assert(
+            $value is string && C\contains_key($enum::getValues(), $value),
+            'Expected a valid value for %s, got %s',
+            static::NAME,
+            (string)$value
+        );
+        return $enum::getValues()[$value as string];
```

Comment:
nitpick: We can avoid this `as` (I try to avoid runtime assertion as they can be a dangerous way to bypass the typechecker) if you use `if (...) throw new UserFacingException(...)` instead of `GraphQL\assert`

This runtime assertion is clearly fine though, so I don't feel too strongly about it.

Reply:
Will update!